src/utils/proxy.js

```diff
@@ -20,14 +22,28 @@ const toReadableStream = require('to-readable-stream')
 
 const edgeFunctions = require('../lib/edge-functions')
 const { fileExistsAsync, isFileAsync } = require('../lib/fs')
+const renderErrorTemplate = require('../lib/render-error-remplate')
 
 const { NETLIFYDEVLOG, NETLIFYDEVWARN } = require('./command-helpers')
 const { createStreamPromise } = require('./create-stream-promise')
 const { headersForPath, parseHeaders } = require('./headers')
 const { createRewriter, onChanges } = require('./rules-proxy')
 
+const decompress = util.promisify(zlib.gunzip)
 const shouldGenerateETag = Symbol('Internal: response should generate ETag')
 
+const formatEdgeFunctionError = (errorBuffer, acceptsHtml) => {
+  const parsedError = JSON.parse(errorBuffer.toString())
+
+  return acceptsHtml
```

Comment:
nit: I think doing an `if` with an early return would make this a lot easier than this ternary expression.